source/Halibut.Tests/Diagnostics/ExceptionReturnedByHalibutProxyExtensionMethodFixture.cs

```diff
@@ -0,0 +1,259 @@
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Halibut.Diagnostics;
+using Halibut.Exceptions;
+using Halibut.ServiceModel;
+using Halibut.Tests.TestServices;
+using Halibut.Tests.Util;
+using Halibut.Transport.Proxy;
+using NUnit.Framework;
+
+namespace Halibut.Tests.Diagnostics
+{
+    public static class ExceptionReturnedByHalibutProxyExtensionMethodFixture
+    {
+        public class WhenGivenA
```

Comment:
No tests around inner exceptions here since the exception containing an inner exception is "just the way it happens to work" rather than explicit behaviour like throwing a `MethodNotFoundHalibutClientException` from a halibut client object.